### PR TITLE
add action related endpoints

### PR DESCRIPTION
**Pull request by Agwen12**
image

image

### CODE REVIEW

**Review by pgorgolew — DISMISSED** [review]

File: src/main/java/org/k8loud/executor/service/ActionService.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    @NotNull
+    private List<ClassElement> getClassElements(ClassPathScanningCandidateComponentProvider scanner, Set<ClassElement> nonAbstractClasses, String fullPackageName) {
+        for (var beanDefinition : scanner.findCandidateComponents(fullPackageName)) {
+            try {
+                Class<?> clazz = Class.forName(beanDefinition.getBeanClassName());
+                if (!clazz.isInterface() && !java.lang.reflect.Modifier.isAbstract(clazz.getModifiers())) {
+
+                    nonAbstractClasses.add(new ClassElement(
+                            clazz.getSimpleName(),
+                            clazz.getPackageName().substring(28),
```
Comment (proman3419) [inline src/main/java/org/k8loud/executor/service/ActionService.java:45]: lmao

File: Makefile [hunk 2]
```diff
@@ -4,7 +4,7 @@ DOCKERHUB_USERNAME=k8loud
 IMAGE_NAME=themis-executor
 # w.x.y.z, one digit value each
 # when tinkering add -<description> suffix
-VER=0.0.4.1
+VER=0.0.4.2
```
Comment (proman3419) [inline Makefile:7]: Change in `manifests/05-deployment.yaml` as well

**Review by proman3419 — CHANGES_REQUESTED** [review]

File: src/main/java/org/k8loud/executor/service/ActionService.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+                    // Handle exceptions if needed
+                }
+            }
+        }
+
+        // If no class is found, return null
+        return null;
+    }
+    private String getFullPackageName(String subPackageName) {
+        return "org.k8loud.executor.actions" + "." + subPackageName;
```
Comment (pgorgolew) [inline src/main/java/org/k8loud/executor/service/ActionService.java:85]: "org.k8loud.executor.actions." + subPackageName;

File: src/main/java/org/k8loud/executor/service/ActionService.java (hunk 1 above)
Comment (Agwen12) [reply src/main/java/org/k8loud/executor/service/ActionService.java:45]: it is what it is xDD

**Review by pgorgolew — APPROVED** [review]

**Review by proman3419 — APPROVED** [review]

**Pull request merged**